src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1641,6 +1650,47 @@ internal override void CreateAsync()
 
             // Create reader thread and send first PSRP message.
             StartReaderThread(_stdOutReader);
+
+            if (_connectionInfo.ConnectingTimeout < 0)
+            {
+                return;
+            }
+
+            // Start connection timeout timer if requested.
+            _connectionTimer = new Timer(
+                callback: (_) => 
+                {
+                    if (_connectionEstablished)
+                    {
+                        return;
+                    }
```

Comment:
Please clarify: if the connection hangs the callback will write the error message  infinitely?

Reply:
Timeout callback is only called once.  See line 1693 `period: Timeout.Infinite`.

`HandleSSHError()` closes the connection.  I'll add a comment to make it more clear.